### ISSUE TITLE
Plugin fails under Maven 3.0.x

### ISSUE THREAD

**Issue by sergei-ivanov** [trustin/os-maven-plugin#5]
Hi,

I have got a test environment that has Maven 3.0.3 as a default version (I use it to test other maven plugins for compatibility), and it turned out that `os-maven-plugin` does not support Maven prior to 3.1 because of missing Aether dependencies.

The error is as follows:
```
$ mvn clean verify
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Detecting the operating system and CPU architecture
[INFO] ------------------------------------------------------------------------
[INFO] os.detected.name: windows
[INFO] os.detected.arch: x86_32
[INFO] os.detected.classifier: windows-x86_32
---------------------------------------------------
constituent[0]: file:/c:/Java/apache-maven-3.0.3/lib/aether-api-1.11.jar
constituent[1]: file:/c:/Java/apache-maven-3.0.3/lib/aether-connector-wagon-1.11.jar
constituent[2]: file:/c:/Java/apache-maven-3.0.3/lib/aether-impl-1.11.jar
constituent[3]: file:/c:/Java/apache-maven-3.0.3/lib/aether-spi-1.11.jar
constituent[4]: file:/c:/Java/apache-maven-3.0.3/lib/aether-util-1.11.jar
constituent[5]: file:/c:/Java/apache-maven-3.0.3/lib/commons-cli-1.2.jar
constituent[6]: file:/c:/Java/apache-maven-3.0.3/lib/maven-aether-provider-3.0.3.jar
constituent[7]: file:/c:/Java/apache-maven-3.0.3/lib/maven-artifact-3.0.3.jar
constituent[8]: file:/c:/Java/apache-maven-3.0.3/lib/maven-compat-3.0.3.jar
constituent[9]: file:/c:/Java/apache-maven-3.0.3/lib/maven-core-3.0.3.jar
constituent[10]: file:/c:/Java/apache-maven-3.0.3/lib/maven-embedder-3.0.3.jar
constituent[11]: file:/c:/Java/apache-maven-3.0.3/lib/maven-model-3.0.3.jar
constituent[12]: file:/c:/Java/apache-maven-3.0.3/lib/maven-model-builder-3.0.3.jar
constituent[13]: file:/c:/Java/apache-maven-3.0.3/lib/maven-plugin-api-3.0.3.jar
constituent[14]: file:/c:/Java/apache-maven-3.0.3/lib/maven-repository-metadata-3.0.3.jar
constituent[15]: file:/c:/Java/apache-maven-3.0.3/lib/maven-settings-3.0.3.jar
constituent[16]: file:/c:/Java/apache-maven-3.0.3/lib/maven-settings-builder-3.0.3.jar
constituent[17]: file:/c:/Java/apache-maven-3.0.3/lib/nekohtml-1.9.6.2.jar
constituent[18]: file:/c:/Java/apache-maven-3.0.3/lib/plexus-cipher-1.4.jar
constituent[19]: file:/c:/Java/apache-maven-3.0.3/lib/plexus-component-annotations-1.5.5.jar
constituent[20]: file:/c:/Java/apache-maven-3.0.3/lib/plexus-interpolation-1.14.jar
constituent[21]: file:/c:/Java/apache-maven-3.0.3/lib/plexus-sec-dispatcher-1.3.jar
constituent[22]: file:/c:/Java/apache-maven-3.0.3/lib/plexus-utils-2.0.6.jar
constituent[23]: file:/c:/Java/apache-maven-3.0.3/lib/sisu-guice-2.9.4-no_aop.jar
constituent[24]: file:/c:/Java/apache-maven-3.0.3/lib/sisu-inject-bean-2.1.1.jar
constituent[25]: file:/c:/Java/apache-maven-3.0.3/lib/sisu-inject-plexus-2.1.1.jar
constituent[26]: file:/c:/Java/apache-maven-3.0.3/lib/wagon-file-1.0-beta-7.jar
constituent[27]: file:/c:/Java/apache-maven-3.0.3/lib/wagon-http-lightweight-1.0-beta-7.jar
constituent[28]: file:/c:/Java/apache-maven-3.0.3/lib/wagon-http-shared-1.0-beta-7.jar
constituent[29]: file:/c:/Java/apache-maven-3.0.3/lib/wagon-provider-api-1.0-beta-7.jar
constituent[30]: file:/c:/Java/apache-maven-3.0.3/lib/xercesMinimal-1.9.6.2.jar
---------------------------------------------------
Exception in thread "main" java.lang.NoClassDefFoundError: org/eclipse/aether/RepositorySystemSession
        at kr.motd.maven.os.DetectExtension.injectSession(DetectExtension.java:115)
        at kr.motd.maven.os.DetectExtension.afterProjectsRead(DetectExtension.java:92)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:273)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:156)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:537)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:196)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:141)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:601)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:290)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:230)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:409)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:352)
Caused by: java.lang.ClassNotFoundException: org.eclipse.aether.RepositorySystemSession
        at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass(SelfFirstStrategy.java:50)
        at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:244)
        at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:230)
        ... 15 more
```

A simple solution would be to include a prerequisite of Maven 3.2.1 (the version currently in use in plugin dependencies). A better solution would be to use Maven compatibility API (if possible) in order to support all Maven 3.x versions until Maven 3.0.x is declared EOL by the Apache team.

Please let me know if you'd like me prepare a pull request.

Kind regards,
Sergei

**Comment by sergei-ivanov** [comment]
Pull request #6 submitted

**Comment by trustin** [comment]
Thank you! I pushed the commit with merge conflict fixed: a42f31bda6d4b94116f309b73c267c3b427f9de4
